### PR TITLE
Trainee preference

### LINKED ISSUE #61: Trainee profile prefences/settings

### PR DESCRIPTION
**Pull request by Christian-Ishimwe**
### Description
This feature enhances the Settings screen by providing users with configurable options for managing their preferences in the application. Users can now customize their profile,  adjust language settings in a simplified interface, with notifications defaulted to false. The focus is on delivering a setting preference screen.

### Description of tasks completed

1. **Profile Management:** Users can navigate to edit their profile
2. **Appearance Settings:** Users can set their theme preference to Light, Dark, or System Default.
Uses a picker component with color-scheme-based styling for a consistent UI experience.
3. **Language Preferences:** Allows users to select their preferred language from the LanguagePicker component.
Language setting is displayed and can be easily updated.
4. **Notifications:** Email and push notification toggles are present but defaulted to false.
5. **Privacy and Security:** Users will be able to view and modify privacy and security settings in the future.

### How this can be tested?

- Clone this PR
- Checkout to the branch `git checkout  ft-preferences`
- install the libraries `npm install`
- Start expo server by `npx expo start`
- Login as a trainee
- on profile icon, there is a dropdown where by you can navigate to preference

### Screenshots
update8
update7

### CODE REVIEW

**Review by Mbabazi12 — APPROVED** [review]

**Review by MugemaneBertin2001 — APPROVED** [review]

**Comment by ncutixavier** [comment]
Please make sure that your implementation match with the figma design

**Review by aimeudoxie — COMMENTED** [review]
- dropdown for changing appearance is not visible even not responsive for IOS
-  unable to scroll to the end of the page make sure you add some bottom margins
-  there is no consistent between colors you used for background and the one we are using for the rest of the app

**Review by aimeudoxie — APPROVED** [review]

File: components/settingPreference/SettingPreference.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+      {/* Language Picker */}
+      <View className={`mb-6 p-4 border-t ${borderColor} rounded-lg flex justify-between`}>
+        <View className="flex-1 mr-4">
+          <Text className={`text-xl font-bold ${textStyle}`}>Language</Text>
+          <Text className={`${textStyle} mt-2 text-lg`}>Language Preferences</Text>
+        </View>
+        <View className="mt-4">
+          <LanguagePicker showFlag={false} />
+        </View>
+      </View>
```
Comment (pacifiquemboni) [inline components/settingPreference/SettingPreference.tsx]: try this code and see if it look the same as the one we have on figma

  {/* Language Picker */}

      <View className={`mb-6 p-4 border-t ${borderColor} rounded-lg  justify-between`}>
       <Text className={`text-xl font-bold ${textStyle}`}>Language</Text>
        <View className="flex-row justify-between  w-full mr-4">
          <Text className={`${textStyle} mt-2 text-lg flex-1`}>Language Preferences</Text>
          <View className={`border  ${borderColor} w-44 flex-1 rounded-lg mt-4`}>
          <LanguagePicker showFlag={false} />
        </View>
        </View>
      </View>

**Review by pacifiquemboni — APPROVED** [review]

**Comment by ncutixavier** [comment]
@Christian-Ishimwe  you should check well the text colors in the ligth mode

File: components/ProfileDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            <Text style={textStyle}>Settings</Text>
+          </TouchableOpacity>
+        </View>
+      </Popover>
+    </View>
+  );
+};
+
+// Define styles
+const styles = StyleSheet.create({
```
Comment (fredshema) [inline components/ProfileDropdown.tsx]: We moved from native styles to tailwind, please use tailwind in your styles

Comment (Christian-Ishimwe) [reply components/ProfileDropdown.tsx]: Resolved

File: app/auth/login/index.tsx [hunk 3]
```diff
@@ -86,7 +86,30 @@ export default function SignInOrganization() {
               
               params.redirect
                 ? router.push(`${params.redirect}` as Href<string | object>)
-                : router.push('/dashboard');
+                : router.push('/dashboard' as Href<string | object>);
+            }
+
+            try {
+              await AsyncStorage.setItem('auth_token', token);
```
Comment (fredshema) [inline app/auth/login/index.tsx]: This is not how the token is stored, change `auth_token` to `authToken`

File: app/auth/login/index.tsx [hunk 4]
```diff
@@ -86,7 +86,30 @@ export default function SignInOrganization() {
...
+              
+              const storedToken = await AsyncStorage.getItem('auth_token');
```
Comment (fredshema) [inline app/auth/login/index.tsx]: Same here, change to `authToken`

**Review by fredshema — APPROVED** [review]

**Review by munezeromicha — APPROVED** [review]

**Review by ncutixavier — APPROVED** [review]

**Pull request closed without merging**